sampleconfig/core.yaml

```diff
@@ -204,6 +204,18 @@ peer:
             # transaction's private data from other peers need to be skipped during the commit time and pulled
             # only through reconciler.
             skipPullingInvalidTransactionsDuringCommit: false
+            # implicitCollectionDisseminationPolicy specifies the dessemination policy for the peer's own implicit collection.
+            # When a peer endorses a proposal that writes to its own implicit collection, below values override the default values
+            # for disseminating private data.
+            # Note that it is applicable to all channels the peer has joined. The implication is that requiredPeerCount has to
+            # be smaller than the number of peers in a channel that has the lowest numbers of peers from the organization.
+            implicitCollectionDisseminationPolicy:
+               # requiredPeerCount defines the minimum number of eligible peers to which the peer must disseminate private data
+               # for its own implicit collection. Default value is 0.
+               requiredPeerCount: 0
```

Comment:
Given that this configuration is applicable for all channels the peer has joined, it might lead to misconfiguration and difficulties in configuration IMO. For example, assume an organization with 4 peers. The organization wants to join 4 channels as well. The peers could join the channels in the following way (depending on a load of each channel and the need for load balancing endorsement requests):
- All 4 peers are part of channel c1 and c4.
- Only peer-1 is part of c2.
- Both peer-2 and peer-3 are part of c3.   
In the end, c1 and c4 have 4 peers. c2 has one peer. c3 has two peers.
peer-1: c1, c2, c4
peer-2: c1, c3, c4
peer-3: c1, c3, c4
peer-4: c1, c4

Given this scenario, how easy or hard for an admin to set these values correctly? In peer-1, the required peer count must be 0 as c2 is only available in peer-1, and so on. As a result, the transaction that gets endorsed on peer-1 for channel c1 would not do any push of pvtData to other peers for the fault-tolerant. I am sure that reading the above configuration itself is confusing but deciding some configuration value based on that is hard too. 

I think that the fault-tolerant property should be handled by the peer implicitly as it knows how many peers have joined a channel, etc... We seem to follow a simple approach and leave the load to the user. This would make the usability a bit harder IMO.

Reply:
@cendhu Thanks. Discussed with Manish last week. There is no precedent in core.yaml for making config channel specific.  Additionally, it is typical for an enterprise to have the same number of peers in each channel, e.g. they frequently have two or three peers in each channel for redundancy, HA, and load balancing. Therefore we concluded that the approach in this PR is a reasonable compromise to keep things simple that should work for most use cases.
There has been a desire to refactor peer configuration in the future, and that would be an opportunity to make configurations channel specific across the board, if there turns out to be demand for it.

MaxPeerCount could potentially be handled by the peer dynamically based on number of available peers as you suggest, but that would not work for RequiredPeerCount, since the whole point of RequiredPeerCount is to fail endorsement in the scenario where other peers become not available.

So, I'm good with the approach in this PR, even though it is sub-optimal.

Reply:
Thanks, Dave. The reasoning for the current approach make sense and the rest can be handled when we refactor peer configuration.